### DISCUSSION TITLE
Ngnix Deployment and Mixed Content Errors

### DISCUSSION THREAD

**Discussion by odedgolden** [aminalaee/sqladmin#156] [Q&A] [answered]
Hi All,

I am using FastAPI + Uvicorn + SQLModel and deployment behind Nginx.

Since my Nginx is configured so that `http` requests are restricted, I get the following errors:
`Mixed Content: The page at '<URL>' was loaded over HTTPS, but requested an insecure script '<URL>'. This request has been blocked; the content must be served over HTTPS.`

This resources of course can be reached when using `https` instead, however, I don't know how to enforce this upgrade to `https` inside `SQLAdmin`.

Assuming I don't want to allow `http`, does anyone have any suggestion on how to solve this?

Thanks a lot! 🙏🏼

**Comment by Uten92** [comment]
Hi @odedgolden,
I have the same problem. Have you found solution?

**Reply by odedgolden** [reply]
Hi @Uten92,
I didn't find a solution.

One option is to to fork the repo and change the http urls to https, and then have it as the dependency instead of the official pip version. In that case you will avoid the Ngnix issue but will have trouble running it locally.

**Comment by aminalaee** [comment]
Hey, I just saw this thread.
I think it might not be possible for you, but Nginx could get HTTPS request but do SSL termination and talk to the backend with HTTP.

If that's not an option, you didn't really explain how you're deploying your project, but you could also use HTTPS with Uvicorn: https://www.uvicorn.org/deployment/#running-with-https

I'm not sure why you would need a change in the Admin side as it should be a few steps before that to handle this.

**Reply by Uten92** [reply]
Thank you, @aminalaee and @odedgolden .

I use nginx as a proxy and I found solution for me in official documentation.

It helped me - using parameter  `--forwarded-allow-ips='*'` and `--proxy-headers` on startup uvicorn https://www.uvicorn.org/deployment/#running-behind-nginx .

**Comment by afparsons** [comment, marked as answer]
Hi everyone, 

I also encountered this problem. I am using:

- Docker service: NGINX reverse proxy
- Docker service: Gunicorn (with uvicorn workers)
- Docker service: PostgreSQL (irrelevant here)
- certificates from Let's Encrypt

This StackOverflow solved my problem: https://stackoverflow.com/a/70485650/4189676

I simply added `--forwarded-allow-ips='*'` to my Gunicorn command. I made no additional changes to my NGINX configuration.
